eth/stagedsync/stage_blockhashes.go

```diff
@@ -0,0 +1,56 @@
+package stagedsync
+
+import (
+	"github.com/ledgerwatch/turbo-geth/common"
+	"github.com/ledgerwatch/turbo-geth/common/dbutils"
+	"github.com/ledgerwatch/turbo-geth/common/etl"
+	"github.com/ledgerwatch/turbo-geth/core/rawdb"
+	"github.com/ledgerwatch/turbo-geth/ethdb"
+)
+
+func extractHeaders(k []byte, v []byte, next etl.ExtractNextFunc) error {
+	if len(k) != 40 {
+		return nil
+	}
+	return next(k, common.CopyBytes(k[8:]), common.CopyBytes(k[:8]))
+}
+
+func SpawnBlockHashStage(s *StageState, stateDB ethdb.Database, quit <-chan struct{}) error {
+	firstHash := common.BytesToHash(s.StageData)
```

Comment:
We stopped actively using `StageData` and `OnLoadCommit`, because it creates some complications. It is easier just to start the whole operation over. As long as it is idempotent (applying it more than once has the same effect as applying once has the same effect), it is a more reliable way to deal with. This means that if you interrupt this stage and restart, it will start from the beginning. I remember you were asking me about it, but I forgot the context, sorry